### PR TITLE
Shave off weavetest package from unnecessary functionality.

### LINKED ISSUE #412: Reorganize imports for test clients
To make a query using a test client, we need app.ResultSet (protobuf) as well as orm.SliceIterator.

Since we don't want to import those packages from weavetest, let's move them. 

SliceIterator (and add a ReverseSliceIterator) into weavetest... Can be imported in orm Tests

ResultSet protobuf along with helpers to and from Models (plus any cleanup of that code) goes into top level weave.

Please check if these suggestions make sense, and feel free to place them somewhere else, but they should be accessible by weavetest.

### PR DESCRIPTION
**Pull request by husio**
This change removes from `weavetest.WeaveRunner` all methods required by
the `weave.ReadOnlyKVStore` interface. There is no reason why the test
runner should implement a store interface. If the user must access the
store in the test, she can do this by directly accessing the store.

I do not know why those methods were added to `weavetest.WeaveRunner`.

This change removes all dependency of the `weavetest` package on
`weave/store` and `weave/app` packages.

resolve #412

### CODE REVIEW

File: weavetest/abci.go [hunk 1]
```diff
@@ -227,81 +222,3 @@ func numBlocks(totalTx, txPerBlock int) int {
 	}
 	return runs
 }
-
-var _ weave.ReadOnlyKVStore = (*WeaveRunner)(nil)
```
Comment (ethanfrey) [inline weavetest/abci.go]: But I use this in benchmarks.
I don't see where this functionality moved to?
#412 specifies moving, not removing, this functionality. Into apps. I am fine with the protobuf stuff gone and much functionality in weave/apps to remove import problems.

Comment (husio) [reply weavetest/abci.go]: I could not find code that is using it. All bnsd benchmarks are passing in this branch.

If you need the database access you can access it directly. Why should `WeaveRunner` implement a database interface to provide an indirect access?

Comment (ethanfrey) [reply weavetest/abci.go]: It was to easily reuse buckets to parse for queries
I see now that benchmarks never query nonce, but rather just increments a local counter, which removed the only query from the benchmarks.

**Comment by ethanfrey** [comment]
Let me push to move this code where it belongs and where requested as part of #412

**Pull request merged**